### PR TITLE
Updated directory structure, readme, and supporting parallel testing on Windows

### PR DESCRIPTION
**Pull request by kn-neeraj**
Change log summary: 

1. Updated directory structure to add self-contained examples directory containing examples such as running the first test, local test, and parallel tests
2. All 3 use cases should work on Unix and Windows environment
3. Updated Readme to provide all the necessary instructions and links to the relevant BrowserStack documentation
4. Minor changes such as more comments to improve readability

Testing :  
All 3 scenarios were tested on - 
1. Python versions - 2.7.9, 3.6, 3.7, 3.8
2. Environments - Windows 10 and Mac
3. Status on the dashboard

### CODE REVIEW

**Review by rishav-csenitjsr — COMMENTED** [review]
Please add `os_version` and `platformName` to the capabilities

File: ios/examples/run-parallel-tests/pavement.py [hunk 1]
```diff
@@ -14,25 +14,23 @@
...
 def run(args):
-    """Run single, local and parallel test using different config."""
-    if args[0] in ('single', 'local'):
-        run_lettuce_test(args[0], args[0])
-    else:
+    if args[0] == 'parallel_tests':
         jobs = []
         for i in range(2):
-            p = multiprocessing.Process(target=run_lettuce_test, args=(args[0], "single", i))
+            p = threading.Thread(target=run_behave_test,args=(i,))
```
Comment (rishav-csenitjsr) [inline ios/examples/run-parallel-tests/pavement.py]: Please change variable names `i` and `p`.

**Review by rishav-csenitjsr — COMMENTED** [review]
Please add a new line at the EOF

File: android/examples/run-first-test/features/steps.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    with AssertContextManager(step):
+        search_element = WebDriverWait(world.browser, 10).until(
+            EC.presence_of_element_located((MobileBy.ACCESSIBILITY_ID, "Search Wikipedia"))
+        )
+        search_element.click()
+
+        search_input = WebDriverWait(world.browser, 30).until(
+            EC.element_to_be_clickable((MobileBy.ID, "org.wikipedia.alpha:id/search_src_text"))
+        )
+        search_input.send_keys(keyword)
```
Comment (rishav-csenitjsr) [inline android/examples/run-first-test/features/steps.py:21]: Do we confirm once, whether we need to add `keyword + "\n"` here or simply `keyword` work fine?

File: android/examples/run-first-test/features/terrain.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from lettuce import before, after, world
+from browserstack.local import Local
+from appium import webdriver
+import os, json
+
+config_file_path = os.path.join(os.path.dirname(__file__), '..', "config.json")
+
```
Comment (rishav-csenitjsr) [inline android/examples/run-first-test/features/terrain.py]: can we print the config file path?

**Review by rishav-csenitjsr — COMMENTED** [review]
We should add `os_version` and `platformName` in the Browserstack doc itself
https://www.browserstack.com/docs/app-automate/appium/getting-started/python/lettuce#Android

**Review by rishav-csenitjsr — COMMENTED** [review]
suggested some minor changes and the changes mentioned in the [PR](https://github.com/browserstack/python-appium-app-browserstack/pull/8).

File: android/examples/run-first-test/features/terrain.py (hunk 3 above)
Comment (kn-neeraj) [reply android/examples/run-first-test/features/terrain.py]: I removed the print statements for now as they were printing twice. We can take it up in next iteration.

File: android/examples/run-first-test/features/steps.py (hunk 2 above)
Comment (kn-neeraj) [reply android/examples/run-first-test/features/steps.py:21]: I checked and the test was working fine without "\n"

**Pull request merged**